sorc/ncep_post.fd/FDLVL.f

```diff
@@ -1,71 +1,45 @@
 !> @file
-!
-!> SUBPROGRAM:    FDLVL       COMPUTES FD LEVEL T, Q, U, V
-!!   PRGRMMR: TREADON         ORG: W/NP2      DATE: 92-12-22       
-!!     
-!! ABSTRACT:
-!!     THIS ROUTINE COMPUTES TEMPERATURE, SPEC. HUM, U WIND COMPONENT,
-!!     AND V WIND COMPONENT ON THE NFD=6 FD LEVELS.  THE
-!!     HEIGHT OF THESE LEVELS (IN METERS) IS GIVEN IN THE 
-!!     DATA STATEMENT BELOW.  THE ALGORITHM PROCEEDS AS 
-!!     FOLLOWS. (AGL IN PARENTHESES)
-!!     
-!!     AT EACH MASS POINT MOVE UP VERTICALLY FROM THE LM-TH (LOWEST
-!!     ATMOSPHERIC) ETA LAYER.  FIND THE ETA LAYERS WHOSE 
-!!     HEIGHT (ABOVE GROUND) BOUNDS THE TARGET FD LEVEL HEIGHT.
-!!     VERTICALLY INTERPOLATE TO GET TEMPERATURE AT THIS FD
-!!     LEVEL.  AVERAGE THE FOUR SURROUNDING WINDS
-!!     TO GET A MASS POINT WIND.  VERTICALLY INTERPOLATE THESE
-!!     MASS POINT WINDS TO THE TARGET FD LEVEL.  CONTINUE THIS
-!!     PROCESS UNTIL ALL NFD=6 FD LEVELS HAVE BEEN PROCESSED.
-!!     MOVE ON TO THE NEXT MASS POINT.  
-!!     
-!!     AVERAGING THE FOUR ABOVE GROUND WINDS TO THE MASS POINT
-!!     WAS FOUND TO SMOOTH THE FIELD AND REDUCE THE OCCURRENCE
-!!     OF POINT PEAK WINDS FAR IN EXCESS OF THE WINDS AT 
-!!     ADJACENT POINTS.  MASS POINT VALUES ARE RETURNED.
-!!     
-!! PROGRAM HISTORY LOG:
-!!   92-12-22  RUSS TREADON
-!!   93-11-23  RUSS TREADON - CORRECTED ROUTINE TO COMPUTE
-!!             FD LEVELS WITH REPECT TO MEAN SEA LEVEL.
-!!   94-01-04  MICHAEL BALDWIN - INCLUDE OPTIONS FOR COMPUTING
-!!                               EITHER AGL OR MSL
-!!   98-06-15  T BLACK - CONVERSION FROM 1-D TO 2-D
-!!   00-01-04  JIM TUCCILLO - MPI VERSION            
-!!   02-01-15  MIKE BALDWIN - WRF VERSION
-!!   11-12-14  SARAH LU - ADD GOCART AEROSOL AERFD
-!!     
-!! USAGE:    CALL FDLVL(ITYPE,TFD,QFD,UFD,VFD)
-!!   INPUT ARGUMENT LIST:
-!!     ITYPE    - FLAG THAT DETERMINES WHETHER MSL (1) OR AGL (2)
-!!                   LEVELS ARE USED.
-!!
-!!   OUTPUT ARGUMENT LIST: 
-!!     TFD      - TEMPERATURE (K) ON FD LEVELS.
-!!     QFD      - SPEC HUM ON FD LEVELS.
-!!     UFD      - U WIND (M/S) ON FD LEVELS.
-!!     VFD      - V WIND (M/S) ON FD LEVELS.
-!!     
-!!   OUTPUT FILES:
-!!     NONE
-!!     
-!!   SUBPROGRAMS CALLED:
-!!     UTILITIES:
-!!
-!!     LIBRARY:
-!!       COMMON   - 
-!!                  LOOPS
-!!                  MASKS
-!!                  OPTIONS
-!!                  INDX
-!!     
-!!   ATTRIBUTES:
-!!     LANGUAGE: FORTRAN
-!!     MACHINE : CRAY C-90
-!!
-!      SUBROUTINE FDLVL(NFD,ITYPE,HTFD,TFD,QFD,UFD,VFD,PFD)
-!      SUBROUTINE FDLVL(ITYPE,TFD,QFD,UFD,VFD,PFD,ICINGFD)
+!> @brief Subroutine that computes FD level T, Q, U, V.
+!>
+!> This routine computes temperature, spec. hum, u wind component,
+!> and v wind component on the NFD=6 FD levels. The 
+!> height of these levels (in meters) is given in the 
+!> date statement below. The alogrithm proceeds as
+!> follows. (AGL in parentheses)
```

Comment:
What is "AGL"?

Reply:
Above ground level.

Reply:
That needs to be in the documentation. ;-)